capture/d2l-capture-producer/src/timeline.js

```diff
@@ -0,0 +1,519 @@
+	 *
+	 * Cannot move past the end of the timeline or any other mark.
+	 * @param {number} pixelsAlongTimelineToMoveTo Point along the timeline to move this mark to.
+	 * @returns {[Cut, Cut]} Cuts that were affected by the move. If the move did not happen, returns null.
+	 * 							[0]: Cut ending at the mark that was stretched or compressed.
+	 * 							[1]: Cut starting at the mark that was stretched or compressed.
+	 */
+	move(pixelsAlongTimelineToMoveTo) {
+		const timeToMoveTo = this.timeline.getTimeFromPixelsAlongTimeline(pixelsAlongTimelineToMoveTo);
+
+		if (timeToMoveTo === this.seconds) return null;
+		else if (timeToMoveTo > this.seconds) { // moving mark forward
+			for (const mark of this.timeline.getMarks()) {
+				// Other mark exists between this mark and the time to move to
+				if (mark !== this && mark.seconds > this.seconds && mark.seconds <= timeToMoveTo) return null;
+			}
+		} else { // moving mark backward
+			for (const mark of this.timeline.getMarks()) {
+				// Other mark exists between this mark and the time to move to
+				if (mark !== this && mark.seconds < this.seconds && mark.seconds >= timeToMoveTo) return null;
+			}
+		}
+
+		const cutStartingAtMark = this.timeline._getCutStartingAtTime(this.seconds);
+
+		if (cutStartingAtMark) {
+			delete this.timeline._cuts[cutStartingAtMark.in];
+
+			cutStartingAtMark.in = timeToMoveTo;
+			this.timeline._cuts[timeToMoveTo] = cutStartingAtMark;
+		}
+
+		const cutEndingAtMark = this.timeline._getCutEndingAtTime(this.seconds);
+
+		if (cutEndingAtMark) {
+			cutEndingAtMark.out = timeToMoveTo;
+		}
+
+		delete this.timeline._marks[this.seconds];
+
+		this.seconds = timeToMoveTo;
+
+		this.timeline._marks[timeToMoveTo] = this;
+
+		return [cutEndingAtMark, cutStartingAtMark];
+	}
+
+	/**
+	 * Removes this mark from the timeline.
+	 * @returns {[Cut, Cut]} Two cuts that were affected by the removal of this mark.
+	 * 							[0]: The cut that ended at this mark, which needed to be extended. If none existed, is null.
+	 * 							[1]: The cut that started at thsi mark, which needed to be removed. If none existed, is null.
+	 */
+	removeFromTimeline() {
+		const cutStartingAtMark = this.timeline._getCutStartingAtTime(this.seconds);
+
+		// Need to remove cut starting here
+		if (cutStartingAtMark) delete this.timeline._cuts[cutStartingAtMark.in];
+
+		const cutEndingAtMark = this.timeline._getCutEndingAtTime(this.seconds);
+
+		// Need to extend cut to the next mark, or the very end
+		if (cutEndingAtMark) {
+			const nextMark = this.timeline._getNextMarkFromTime(this.seconds);
+
+			const outSeconds = nextMark ? nextMark.seconds : this.timeline.durationSeconds;
+
+			cutEndingAtMark.out = outSeconds;
+		}
+
+		delete this.timeline._marks[this.seconds];
+
+		return [cutEndingAtMark, cutStartingAtMark];
+	}
+}
+
+class Timeline {
+	constructor(durationSeconds, widthPixels, cuts = [], zoomMultiplier = 1, pixelsAlongTimelineToZoomAround = Math.round(widthPixels / 2)) {
```

Comment:
Also consider making the argument an object here?
```suggestion
	constructor({ durationSeconds, widthPixels, cuts = [], zoomMultiplier = 1, pixelsAlongTimelineToZoomAround = Math.round(widthPixels / 2)} = {}) {
```